### DISCUSSION TITLE
ibis throws error on geometry column

### DISCUSSION THREAD

**Discussion by carlsunderman** [ibis-project/ibis#7823] [Q&A] [answered]
I have a column of type geometry in mssql server called Shape.  When I try to read the table, I receive an error, but when I connect to a table without the geom column, I receive the data.  Is there a way to either omit the geom column or cast it in the initial connection?

**Comment by gforsyth** [comment]
Hey @carlsunderman -- can show us the schema for the table in question using `tsql` or similar?  There isn't `geospatial` support in the `mssql` backend in Ibis right now, but that shouldn't prevent Ibis from loading a table that has a `geometry` column.

Also, if you can paste the traceback that will also help.

**Comment by carlsunderman** [comment]
I think this should work
https://justpaste.it/89o8r

the data is in WKB and I normally need to do ST_AsText when selecting with sqlalchemy

**Comment by carlsunderman** [comment]
SAWarning: Omitting index key for (Shape), key covers omitted columns.
  self.inspector.reflect_table(table, table.columns)

OperationalError: (pymssql._pymssql.OperationalError) (2809, b"The request for procedure 'GLB_SEISMIC_2DLINES_API' failed because 'GLB_SEISMIC_2DLINES_API' is a table object.DB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\n")
[SQL: EXEC sp_describe_first_result_set @tsql = %(query)s]
[parameters: {'query': 'sde.[GLB_SEISMIC_2DLINES_API]'}]



but when i query a table without the shape column, it loads and like I said, if I connect via sqlalchemy and ST_AsText the shape column it loads there too

**Reply by gforsyth** [reply]
Ah, we've just cut over on the `master` branch to using `pyodbc` instead of `pymsssql`.  If you can try out a recent commit from `master` it may resolve this.

**Reply by carlsunderman** [reply]
same issue after the update

Hope its readable
https://justpaste.it/andmn

**Reply by gforsyth** [reply]
hmm, I do not know what mssql is talking about.  I wonder what a "table object" is if not a table.

**Comment by carlsunderman** [comment]
is there a way with ibis to omit or [SHAPE.]STAsText() this column on connecting? 

[SHAPE.]STAsText() works with create_engine

using create_engine, without the above, I get the  following pyodbc error
 ODBC SQL type-151 is not yet supported. columns-index=1 type=-151, 'HY106

**Reply by gforsyth** [reply, marked as answer]
there is not.  :grimacing:

**Comment by carlsunderman** [comment]
no problem, for this type of data I can use other methods then pass to ibis, thank you for your help
